code/game/turfs/simulated/floor/asteroid.dm

```diff
@@ -0,0 +1,322 @@
+/**********************Asteroid**************************/
+
+/turf/simulated/floor/plating/asteroid
+	name = "asteroid sand"
+	baseturf = /turf/simulated/floor/plating/asteroid
+	icon_state = "asteroid"
+	icon_plating = "asteroid"
+	var/environment_type = "asteroid"
+	var/turf_type = /turf/simulated/floor/plating/asteroid //Because caves do whacky shit to revert to normal
+	var/dug = 0       //0 = has not yet been dug, 1 = has already been dug
+	var/sand_type = /obj/item/stack/ore/glass
+	var/floor_variance = 20 //probability floor has a different icon state
+
+/turf/simulated/floor/plating/asteroid/New()
+	var/proper_name = name
+	..()
+	name = proper_name
+	if(prob(floor_variance))
+		icon_state = "[environment_type][rand(0,12)]"
+
+/turf/simulated/floor/plating/asteroid/burn_tile()
+	return
+
+/turf/simulated/floor/plating/asteroid/ex_act(severity, target)
+	switch(severity)
+		if(3.0)
+			return
+		if(2.0)
+			if(prob(20))
+				gets_dug()
+		if(1.0)
+			gets_dug()
+
+/turf/simulated/floor/plating/asteroid/attackby(obj/item/W, mob/user, params)
+	//note that this proc does not call ..()
+	if(!W || !user)
```

Comment:
TG does this differently, but I'm sure it can be refactored later. As long as it's functionally the same...